clients/admin-ui/src/features/system/VendorSelector.tsx

```diff
@@ -130,165 +132,194 @@ const VendorSelector = ({
-                // eslint-disable-next-line @typescript-eslint/no-unused-vars
-                formatCreateLabel={(_value) => null}
-                chakraStyles={{
-                  container: (provided) => ({
-                    ...provided,
-                    flexGrow: 1,
-                    backgroundColor: "white",
-                  }),
-                  option: (provided, state) => ({
-                    ...provided,
-                    background:
-                      state.isSelected || state.isFocused ? "gray.50" : "unset",
-                  }),
-                  menu: (provided) => ({
-                    ...provided,
-                    visibility: hasVendorSuggestions ? "visible" : "hidden",
-                  }),
-                  dropdownIndicator: (provided) => ({
-                    ...provided,
-                    display: "none",
-                  }),
-                  indicatorSeparator: (provided) => ({
-                    ...provided,
-                    display: "none",
-                  }),
-                }}
-                components={{ Option: CustomDictOption }}
-              />
-              <Text
-                aria-hidden
-                style={{
-                  position: "absolute",
-                  backgroundColor: "transparent",
-                  borderColor: "transparent",
-                  marginTop: "5.52px",
-                  marginLeft: "13px",
-                  pointerEvents: "none",
-                  fontSize: "14px",
-                  zIndex: 1,
-                }}
-              >
-                <span style={{ color: "transparent" }}>{searchParam}</span>
-                {hasVendorSuggestions ? (
-                  <span style={{ color: "#824EF2" }}>
-                    {suggestions[0].label.substring(searchParam.length)}
-                  </span>
-                ) : null}
-              </Text>
-            </Flex>
-            <ErrorMessage
-              isInvalid={isInvalid}
-              message={meta.error}
-              fieldName={fieldsSeparated ? "vendor_id" : "name"}
-            />
-          </VStack>
-        </FormControl>
-        <CompassButton
-          active={!!values.vendor_id || hasVendorSuggestions}
-          disabled={!values.vendor_id || dictSuggestionsState === "showing"}
-          onRefreshSuggestions={() => onVendorSelected(values.vendor_id)}
+  const handleBlur = (event: React.FocusEvent) => {
+    field.onBlur(event);
+    if (searchParam) {
+      setValue(searchParam);
+    }
+    setTouched(
+      {
+        ...touched,
+        name: true,
+      },
+      // only validate if nothing is typed in the select's search input;
+      // this prevents incorrectly showing a "required field" error after
+      // the input turns into a text input while the name is valid
+      !searchParam
+    );
+  };
+
+  const typeaheadSelect = (
+    <FormControl isInvalid={isInvalid} isRequired width="100%">
```

Comment:
what would it take for this to use the existing `CustomCreatableSelect` that exists already? it's fine to say this component is just too specialized for that! if it is, it'd be good to have a comment explaining why we aren't reusing the existing component

Reply:
I think this one is too specialized, it needs specific handling for:

- when to show the menu
- when to show the clear indicator
- blurring
- tab completion
- the value of the inner input
- its own custom option

I'll add a comment about it.